src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -562,4 +568,32 @@ public void deleteAllByExerciseId(Long exerciseId, boolean deleteBuildPlan, bool
             delete(participation.getId(), deleteBuildPlan, deleteRepository);
         }
     }
+
+    @Transactional(readOnly = true)
+    public List<Participation> findByExerciseIdForTutorDashboard(Long id) {
+        List<Participation> participationList = this.findByExerciseId(id);
+
+        for (Participation participation : participationList) {
```

Comment:
The code below looks very inefficient. Please first describe in the JavaDoc comment what you actually want to achieve and then let's brainstorm how we could optimize all these database calls preferably into one or two calls.

Reply:
Also I don't like the exception handling. This is typically quite expensive. Instead try to avoid such exceptions

Reply:
I  am actually thinking to remove this code and rewrite it, in the end is needed only to provide statistics for the course dashboard, we probably can return just the numbers (and thus we need only to `count` on the db)

Reply:
I was indeed able to completely remove this method, and using some `count` in the repository, definitely improving performance!